### DISCUSSION TITLE
Setting per-asset resampling?

### DISCUSSION THREAD

**Discussion by jessjaco** [gjoseph92/stackstac#210] [General]
I'm looking for a way to set resampling for specific assets / bands independently. I need this, for example, when dealing with Landsat data that has a categorical QA band (which I need to open with nearest resampling) and also some continuous spectral data that I would like to apply some smoothing to (e.g. bilinear resampling).

I took a cursory look through the code, and it seems like this would be fairly easy to implement, as [asset_table_to_reader_and_window](https://github.com/gjoseph92/stackstac/blob/7836a363b2609216890b0e20e289f8ff71cf320d/stackstac/to_dask.py#L111) creates a reader for each asset anyway.

Is there any interest in this?

**Comment by gjoseph92** [comment]
@jessjaco, great question. The example of a QA band versus continuous spectral data is a perfect motivating example for this.

In general, I'd say yes, there's interest in this. I'd be even _more_ interested if there were some way stackstac could infer this automatically from the STAC metadata. If users had to pass a tuple or dict or something for `resampling`, that would be pretty cumbersome, and I doubt it would get as much use. It also would still expose people to subtle errors under default behavior like what you probably encountered, where discrete data gets resampled into meaningless values.

I'm not sure what STAC metadata we'd use to determine this. https://github.com/stac-extensions/classification seems like an option, but I'm not sure how widely used it is. Anything else you can think of?

As far as how to implement this, https://github.com/gjoseph92/stackstac/pull/202 is a good recent example of threading a little bit of STAC metadata all the way through to dataset-opening. I imagine a PR to do this for resampling would look very similar.

**Comment by jessjaco** [comment]
Yes, that all makes sense. I will think about the metadata pathway. The only thing is that there are often multiple resamplers that would be appropriate with continuous data. I often use nearest neighbor, but in this case I need something different.

For reference, [odc.stac.load](https://odc-stac.readthedocs.io/en/latest/_api/odc.stac.load.html) does implement per-band resampling with a list or dict, though the documentation doesn't say how to use them (I learned that e.g. `{'qa_pixel':'nearest', '*':'bilinear'}` is an appropriate call for my use case). 

PS If you're wondering why I don't use odc.stac.load it's because [stackstac error handling is more robust](https://github.com/opendatacube/odc-stac/issues/122) and I'm trying to decide which is easier to adapt.
